sui_core/src/consensus_client.rs

```diff
@@ -193,6 +197,13 @@ impl ConsensusClient {
                         // it (the core is idempotent).
                         fp_bail!(SuiError::StorageError(e));
                     }
+                    // The authority missed some consensus outputs and needs to sync.
+                    Err(SuiError::RequireSyncWithConsensus) => {
```

Comment:
Not a great fan of communicating non Error control messages in Err. Since this function has no return I would define an enum with a variant "Get past sequence from X" and use that?

Reply:
me neither actually, fixed